pdm/models/repositories.py

```diff
@@ -124,6 +131,30 @@ def find_candidates(
                 key=lambda c: c.version,
                 reverse=True,
             )
+
+            if not sorted_cans:
+                self.environment.project.core.ui.echo(
+                    "Could not find any matching candidates even when considering "
+                    "pre-releases",
+                    verbosity=termui.DETAIL,
+                    fg="yellow",
+                )
+
+        def print_candidates(title: str, candidates: List[Candidate]) -> None:
+            self.environment.project.core.ui.echo(
+                title, verbosity=termui.DETAIL, fg="blue"
+            )
+            with self.environment.project.core.ui.indent("  "):
```

Comment:
I prefer this message to all go to log file. But the length should be limited, because a project could have ~50 candidates and the log content grows quickly.